src/v0/destinations/mp/util.js

```diff
@@ -173,6 +173,35 @@ const removeDuplicateMetadata = (mergedBatches) => {
   });
 };
 
+/**
+ * Builds UTM parameters from a campaign object.
+ *
+ * @param {Object} campaign - The campaign object containing the campaign details.
+ * @returns {Object} - The object containing the UTM parameters extracted from the campaign object.
+ *
+ * @example
+ * const campaign = {
+ *   name: 'summer_sale',
+ *   source: 'newsletter',
+ *   medium: 'email'
+ * };
+ * { utm_campaign: 'summer_sale', utm_source: 'newsletter', utm_medium: 'email' }
+ */
+const buildUtmParams = (campaign) => {
+  const utmParams = {};
+  if (campaign) {
```

Comment:
Make sure to add a check if `campaign` is a defined object. The current function will fail if `campaign` is an Array or some other truthy value example:  `valid number, string, boolean true`

Reply:
added the isObject check